source/Model.js

```diff
@@ -445,6 +448,12 @@ function testUsername(username, domain) {
   });
 }
 
+// Make sure that the caller id is sent to the API alongside the sharing token
+// by replacing the client separator with the one expected by the API
+function formatSharingCallerId(sharing) {
```

Comment:
I've questions ;)

In spirit, this method seems to receive 'www-urlencoded' strings and decode them to literal strings. Seen like that, it seems to me that we have to worry about more than just the plus ('+'), right?  

This link 

https://stackoverflow.com/questions/12042592/decoding-url-parameters-with-javascript#16587536

explains why we will still need to special case the '+' though. 

Or - alternatively - I didn't see the unescape (i.e. decodeURIComponent) and it is being done earlier. Did you test with using '%20' instead of '+' as well? The spec now doesn't really use the '+' anymore - and there might be other chars that we would need to escape.